dashboard/app/models/user.rb

```diff
@@ -838,7 +865,7 @@ def User.track_proficiency(user_id, script_id, level_id)
 
   # returns whether a new level has been completed and asynchronously enqueues an operation
   # to update the level progress.
-  def track_level_progress_async(script_level:, level:, new_result:, submitted:, level_source_id:)
+  def track_level_progress_async(script_level:, level:, new_result:, submitted:, level_source_id:, pairings:)
```

Comment:
They're named parameters. It's basically a hash, and without default values
they're all required but can be supplied in any order.

On Wed, Jun 29, 2016 at 1:49 PM, ashercodeorg <notifications@github.com>
wrote:

> In dashboard/app/models/user.rb
> <https://github.com/code-dot-org/code-dot-org/pull/5556#discussion_r69024763>
> :
>
> > @@ -838,7 +865,7 @@ def User.track_proficiency(user_id, script_id, level_id)
> >
> >    # returns whether a new level has been completed and asynchronously enqueues an operation
> >    # to update the level progress.
> > -  def track_level_progress_async(script_level:, level:, new_result:, submitted:, level_source_id:)
> > +  def track_level_progress_async(script_level:, level:, new_result:, submitted:, level_source_id:, pairings:)
>
> Basic Ruby Q: What do the colons at the end of the variable denote?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/5556/files/9fd5d37b0126f1cef6e80e4b86fd41f3c7ddd543#r69024763>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AIBBRPPTxmRVlwymCKOzqU_MW_Goj2Zfks5qQtplgaJpZM4Gmr2B>
> .
>